### PR TITLE
introduce processorName option for KSP.

### PR DESCRIPTION
**Pull request by neetopia**
This option will allow users to specify certain processors(s) to be
executed, if not specified, all processors found in apclasspath will be
executed.

### CODE REVIEW

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspExtension.kt [hunk 1]
```diff
@@ -33,6 +33,8 @@ open class KspExtension {
         apOptions.put(k, v)
     }
 
+    open var processorName: List<String> = emptyList()
```
Comment (gavra0) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspExtension.kt:36]: Rename to `processorNames`, considering there are 1+ you can specify?

File: gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt [hunk 2]
```diff
@@ -165,6 +165,9 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             kspExtension.apOptions.forEach {
                 options += SubpluginOption("apoption", "${it.key}=${it.value}")
             }
+            kspExtension.processorName.forEach {
+                options += SubpluginOption("processorName", "$it")
```
Comment (gavra0) [inline gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:169]: Is this a task input? Also, consider adding an integration test in which you modify processor names and assert that task is not up-to-date.

Comment (neetopia) [reply gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt:169]: done, thanks for pointing this out!

**Comment by ting-yuan** [comment]
Setting the processor names in KspExtension makes it hard to track in Gradle tooling, which would need to depend on KSP implementation, instead of the common approach of reading configuration / dependencies. OTOH, if users only need certain processors to be loaded, they should probably not include other dependencies in the beginning. Could we remove the support in Gradle, i.e., remove the option in KspExtension? So as to keep it only available in command line for now.

**Review by ting-yuan — APPROVED** [review]

**Pull request merged**